### PR TITLE
Node SDK

### PR DESCRIPTION
**Pull request by tgroutars**

### CODE REVIEW

**Review by KevinTss — COMMENTED** [review]
Make a naive review if it can help 😄

File: packages/node/src/UnrevealedClient.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  private _featureAccesses: Record<string, FeatureAccess> = {};
+  private readonly _apiKey: string;
+  private readonly _apiUrl: string;
+  private _logger = new Logger();
+  private _readyState: ReadyState = ReadyState.UNINITIALIZED;
+  private _connectionPromise: Promise<void> | null = null;
+
+  constructor({ apiKey, apiUrl = SSE_API_URL }: UnrevealedClientOptions) {
+    this._apiKey = apiKey;
+    this._apiUrl = apiUrl;
```
[the remark this answers is not in the document]
Comment (tgroutars) [reply packages/node/src/UnrevealedClient.ts:39]: This is more for me for local development, I think I'll remove it from the `UnrevealedClientOptions` types so people don't use it without knowing what they're doing

File: packages/node/src/UnrevealedClient.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  UNINITIALIZED,
+  CONNECTING,
+  READY,
+  CLOSED,
+}
+
+const RETRY_INTERVAL_MS = 2000;
+const RETRY_MAX_ATTEMPTS = 5;
+
+export interface UnrevealedClientOptions {
```
[the remark this answers is not in the document]
Comment (tgroutars) [reply packages/node/src/UnrevealedClient.ts:23]: I'm re-exporting this from the package so users can use it in their code

File: packages/node/src/Logger.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export class Logger {
+  log(message: string) {
+    console.log(`[INFO] unrevealed: ${message}`);
+  }
+
+  error(message: string) {
+    console.log(`[ERROR] unrevealed: ${message}`);
```
[the remark this answers is not in the document]
Comment (tgroutars) [reply packages/node/src/Logger.ts:7]: I was using console.error before I extracted it here, it was an oversight and I'll change it back

File: packages/node/src/UnrevealedClient.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  private async _connectRecursive(attempt: number = 1) {
+    this._readyState = ReadyState.CONNECTING;
+
+    try {
+      await this._connect();
+      this._readyState = ReadyState.READY;
+      this._logger.log('Connection to Unrevealed established');
+    } catch (err) {
+      if (err instanceof UnauthorizedException) {
+        this._logger.error('Unauthorized, please check your API key');
```
[the remark this answers is not in the document]
Comment (tgroutars) [reply packages/node/src/UnrevealedClient.ts:82]: Good catch, I should close here

File: packages/node/src/errors.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export class UnauthorizedException extends Error {}
```
[the remark this answers is not in the document]
Comment (tgroutars) [reply packages/node/src/errors.ts:1]: I can throw it somewhere and check using `instanceof` in a catch block

File: packages/node/test.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+const message = 'cocou';
```
[the remark this answers is not in the document]
Comment (tgroutars) [reply packages/node/test.js:1]: Hahaha that wasn't supposed to be committed

File: packages/node/.eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{
+  "env": {
+    "browser": true,
```
[the remark this answers is not in the document]
Comment (tgroutars) [reply packages/node/.eslintrc.json:3]: It's for node, so yeah it should be `node` not `browser` 👌

**Pull request merged**